A88-xds-data-error-handling.md

```diff
@@ -0,0 +1,451 @@
+NACK from client | true | false | `OnError(status)`, Ignore | `OnAmbientError(Status)`, Ignore
+NACK from client | true | true  | `OnError(status)`, Ignore | `OnResourceChanged(Status)`, Drop existing resource and fail RPCs
+Resource timeout | false | (any) | `OnResourceDoesNotExist()`, Fail data plane RPCs | `OnResourceChanged(Status(NOT_FOUND))` (status will be UNAVAILABLE if `resource_timer_is_transient_error` server feature is present), Fail data plane RPCs
+LDS or CDS resource deletion from server | true | false | `OnResourceDoesNotExist()`, Drop existing resource and fail RPCs (watcher notification skipped if "ignore_resource_deletion" server feature is present) | `OnAmbientError(Status(NOT_FOUND))`, Ignore
+LDS or CDS resource deletion from server | true | true | `OnResourceDoesNotExist()`, Drop existing resources and fail RPCs (watcher notification skipped if "ignore_resource_deletion" server feature is present) | `OnResourceChanged(Status(NOT_FOUND))`, Drop existing resource and fail RPCs
+[xRFC TP3] error with status NOT_FOUND or PERMISSION_DENIED | false | (any) | N/A | `OnResourceChanged(status)`, Fail data plane RPCs
+[xRFC TP3] error with status NOT_FOUND or PERMISSION_DENIED | true | false | N/A | `OnAmbientError(status)`, Ignore
+[xRFC TP3] error with status NOT_FOUND or PERMISSION_DENIED | true | true | N/A | `OnResourceChanged(status)`, Drop existing resource and fail RPCs
+[xRFC TP3] error with other status | false | (any) | N/A | `OnResourceChanged(status)`, Fail data plane RPCs
+[xRFC TP3] error with other status | true | (any) | N/A | `OnAmbientError(status)`, Ignore
+
+### Propagating Status to Data Plane RPCs
+
+As noted in [gRFC A54], the set of status codes that gRPC should generate
+for RPCs is intentionally limited.  In particular, INVALID_ARGUMENT and
+NOT_FOUND are two of the codes that are reserved for application use
+and should never be generated by gRPC itself.  Implementors should not
+blindly propagate status codes from the XdsClient watcher to data plane
+RPCs; instead, they should be careful to set status codes carefully on
+data plane RPCs on both the client and server side.  In most cases, it
+is expected that UNAVAILABLE is an appropriate code for data plane RPCs.
+
+### XdsClient Metric Labels
+
+[gRFC A78] defines a set of XdsClient metrics to be exported via
+OpenTelemetry.  One of the metric labels defined there is the
+`grpc.xds.cache_state` label, which is used for the
+`grpc.xds_client.resources` metric.  In keeping with the new CSDS client
+state introduced in [xRFC TP3], we will add two new possible values for
+this label:
+- `received_error`: The server sent an error for this resource.
+- `received_error_but_cached`: There is a version of this resource
+  cached, but the server subsequently sent an error for this resource.
+
+By default, the second value will be seen; if the `fail_on_data_errors`
+server feature is present, then the second value will be seen.
```

Comment:
Both default and condition mention `second value`, I believe you intended default to be the first value.

Reply:
Good catch, fixed.